subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedBeanAnnotationHandler.java

```diff
@@ -59,15 +61,19 @@ public void visitPropertyValue(String propertyName, PropertyValue value, Propert
 
     private static final Set<Class<?>> SUPPORTED_KEY_TYPES = new HashSet<>(Arrays.asList(String.class, Integer.class, Enum.class));
 
+    private static String getSupportedKeyTypes() {
+        return SUPPORTED_KEY_TYPES.stream().sorted(Comparator.comparing(Class::toString)).map(cls -> "'" + cls.getSimpleName() + "'").collect(Collectors.joining(", "));
+    }
```

Comment:
🧐 It's good that you thought of sorting the entries! It'd be even better to use a data structure that doesn't need sorting. For static collections we always use Guava's `ImmutableSet` and friends. These collections are immutable, and they retain insertion order (even without the extra overhead of a `LinkedHashSet`), so you can pick what order the elements will appear. `ImmutableCollection`s have nice static factory methods and builders that allow them to be constructed with less boilerplate, too.

Reply:
Great, thanks for pointing that out!